pkg/handler/eval_cache.go

```diff
@@ -70,13 +92,43 @@ func (ec *EvalCache) GetByTags(tags []string) []*entity.Flag {
 			}
 		}
 	}
+	return results
+}
 
-	values := make([]*entity.Flag, 0, len(results))
-	for _, f := range results {
-		values = append(values, f)
+func (ec *EvalCache) getByTagsALL(tags []string) map[uint]*entity.Flag {
+	results := map[uint]*entity.Flag{}
+	cache := ec.cache.Load().(*cacheContainer)
+
+	for i, t := range tags {
+		fSet, ok := cache.tagCache[t]
+		if !ok {
+			// clear as no flags
+			results = map[uint]*entity.Flag{}
+			continue
+		}
+
+		if i == 0 {
+			// store all the flags
+			for fID, f := range fSet {
+				results[fID] = f
+			}
```

Comment:
you can just do `results = fSet`

Reply:
nvm, I think yours is better, since we are deleting the keys from results on the fly, better keep a shallow clone of the map.